src/components/CustomChannelComponent.tsx

```diff
@@ -238,9 +242,16 @@ export function CustomChannelComponent() {
   const welcomeMessageTimeStamp =
     lastWelcomeMessageCreatedAt ?? firstMessageCreatedAt;
 
+  const resetReqCounter = useRef(0);
+
   return (
     <Root height={'100%'} isStaticReplyVisible={isStaticReplyVisible}>
-      <ChannelUI
+      <GroupChannelUI
+        onChannelFetchFailed={() => {
+          if (resetReqCounter.current > 5) return;
+          resetReqCounter.current += 1;
+          resetSession();
+        }}
```

Comment:
I want to understand the case where onChannelFetchFailed is called multiple times such that this code is executed multiple times. Could you explain the case?

Reply:
> fetch channel -> not found channel -> onChannelFetchFailed -> resetSession -> channel created -> channel deleted -> fetch channel -> not found channel -> onChannelFetchFailed -> ... loop